src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/probation/ProbationUnmergeEventProcessor.kt

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.hmpps.personrecord.message.processors.probation
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.personrecord.client.model.sqs.domainevent.DomainEvent
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType.DELIUS
+import uk.gov.justice.digital.hmpps.personrecord.service.EventKeys
+import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.UNMERGE_MESSAGE_RECEIVED
+
+@Component
+class ProbationUnmergeEventProcessor(
+  val telemetryService: TelemetryService,
+  val personRepository: PersonRepository,
+) : BaseProbationEventProcessor() {
```

Comment:
I know this hasn't been introduced in this PR but I am not over keen on using inheritance like this. `getProbationCase` could just as easily be in a service, which would make it available for reuse should anything which isn't a ProbationEventProcessor wish to use it.

Reply:
There are loads of articles about favouring composition over inheritance around, and I am sure you are familiar with the pros and cons. I have generally found that it can be useful in tests (where refactoring and new development is rare) and restrictive in the `main` source tree where you need to be flexible and esponse.

Reply:
Discussed with mark, potentially need to do something around the base event classes down the line as they could cause issue in the future